src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -0,0 +1,59 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Base classes for the power manager."""
+
+from __future__ import annotations
+
+import abc
+import dataclasses
+import enum
+
+from ...timeseries import Power
+
+
+@dataclasses.dataclass(frozen=True)
+class Proposal:
+    """A proposal for a battery to be charged or discharged."""
+
+    source_id: str
```

Comment:
Especially the source id should be documented imo

Reply:
done in this commit: https://github.com/frequenz-floss/frequenz-sdk-python/pull/692/commits/1e7f54bb7be45f7f4ddb3170cfe16d501530ef3f